src/rdkafka_buf.h

```diff
@@ -195,8 +195,6 @@ rd_tmpabuf_write_str0 (const char *func, int line,
                                 "expected %"PRIusz" bytes > %"PRIusz    \
                                 " remaining bytes",                     \
                                 __len0, rd_kafka_buf_read_remain(rkbuf)); \
-                        (rkbuf)->rkbuf_err = RD_KAFKA_RESP_ERR__BAD_MSG; \
```

Comment:
i'm not going to bother understanding why you took this out, I assume you know what you're doing.

Reply:
parse_fail() already sets it